### PR TITLE
Add copy to clipboard option for project/executable source columns

### LINKED ISSUE #1309: Add ability to copy Source file path in dashboard resources page
With #1232 the file path is no longer available in the view. 

Add the ability to copy the file path:
image
and a "Copy" option where it would copy the path to clipboard

### PR DESCRIPTION
**Pull request by adamint**
It looks like this now:
image

I also added this for executables, which removed the gray arg subtext so that the entire command appeared the same, as GridValue accepted only strings (I did not understand why there was subtext to begin with).

closes #1309

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1332)

### CODE REVIEW

**Comment by tlmii** [comment]
> I did not understand why there was subtext to begin with

It was there to make it easier to visually differentiate the different pieces of information we were cramming into one column (executable vs args vs working directory). With the different types of data in a single column (both for one resource type, like here, and across different resource types) visual scanning was difficult.

It's not strictly necessary, of course, but if we're going to remove it we should consider what else we can do to make the information more easily scannable.

**Comment by adamint** [comment]
> It was there to make it easier to visually differentiate the different pieces of information we were cramming into one column

Right, the executable path + command seemed to me like two parts of the same information (what is being executed). Since working directory is now on a second line, do you still have that concern? @tlmii

**Comment by tlmii** [comment]
> Since working directory is now on a second line?

Working directory was on a second line when we originally crammed everything together (#987) - command and args were on one line (but differentiated by color), working directory was set off on the second line _and_ labeled.
 
> Right, the executable path + command seemed to me like two parts of the same information (what is being executed). 

I agree that, at one level, executable + args are two parts of one concept (because of course together they form the full command). But from a "what am I looking for in the column?" perspective, a common thing is to look for just the executable name, and visually parsing that off the full command is easier when it is set off somehow. It's the same concept as how commands and args get colored differently in powershell for example:

image

I'm not saying this visual scanning is the only important thing, nor that this is the only way to solve it. But it does follow a common pattern. Perhaps we can loop in @leslierichardson95 to get her take.

**Comment by adamint** [comment]
Ah, apologies for the misunderstanding. We are able to keep the current behavior, it just requires a small change to `GridValue` to allow for fragments. I can go ahead and make that

**Comment by tlmii** [comment]
> Ah, apologies for the misunderstanding. 

I think it is a valuable conversation to have - there's more than one way to accomplish these goals and you never know what'll come out of the discussions!

**Comment by adamint** [comment]
Alright, added the option to have a fragment after the value.
image

**Comment by adamint** [comment]
@tlmii @drewnoakes for re-review

**Review by tlmii — APPROVED** [review]
I'm not 100% sure about the `ContentAfterValue` concept. But given how this component works, with the FluentHighlighter for the main value, if you did a normal fragment for the whole of Value's display, then you'd need to handle turning off highlighting, too, which just seems like too much. So at least for now I think this is a good compromise. We can always revisit if necessary.

**Review by smitpatel — APPROVED** [review]

**Comment by JamesNK** [comment]
Could this also be applied to containers? The button could copy the image name and tag, e.g. `postgres:latest`.

Right now it feels weird that some rows in a grid have a copy button in that column, and others don't.

File: src/Aspire.Dashboard/Resources/Columns.resx [hunk 1]
```diff
@@ -136,6 +136,9 @@
   <data name="SourceColumnDisplayContainerCommand" xml:space="preserve">
     <value>Command: {0}</value>
   </data>
+  <data name="SourceColumnSourceCopyFullPathToClipboard" xml:space="preserve">
+    <value>Copy File Path to Clipboard</value>
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Resources/Columns.resx:140]: I really don't like the casing here. It's just a sentence. Why is everything captalized? (I know you followed an existing pattern in other tooltips. I also think they're wrong).

What I'd like:
```suggestion
    <value>Copy file path to clipboard</value>
```

Also update the other copy tooltips to be cased like a regular sentence.

Comment (JamesNK) [reply src/Aspire.Dashboard/Resources/Columns.resx:140]: For example, in GitHub tooltips have normal sentence capitalization:
image

Comment (tlmii) [reply src/Aspire.Dashboard/Resources/Columns.resx:140]: Yeah our style guide basically says sentence casing almost everywhere. We (I) did not do that consistently earlier.

**Comment by tlmii** [comment]
> Could this also be applied to containers? The button could copy the image name and tag, e.g. `postgres:latest`.
> 
> Right now it feels weird that some rows in a grid have a copy button in that column, and others don't.

I think at some point we're going to have to pull these columns back apart somehow. Having multiple different pieces of info (both within a single resource type and across resource types) within a column leads us down these rabbit trails like this. Hopefully we get some good feedback on it.

**Review by JamesNK — APPROVED** [review]

**Pull request merged**